Examples/graph-toy.lisp

```diff
@@ -104,15 +104,15 @@
         (xval-count (max-xvals frame)))
     (when (>= val-array-len (* xval-count 2))
       (loop for i from 0 to (- xval-count 1)
-            for j from (- val-array-len xval-count)
-            do (setf (elt (val-array frame) i) (elt (val-array frame) j)))
+         for j from (- val-array-len xval-count)
+         do (setf (elt (val-array frame) i) (elt (val-array frame) j)))
       (setf (fill-pointer (val-array frame)) xval-count))
-    (redisplay-frame-pane frame 'main-display)
-    (incf (last-yval frame) (- (random 9) 4))
     (when (member (frame-state frame) '(:enabled :shrunk))
```

Comment:
So the current implementation is like:
1. update the frame state
2. check if frame state is valid
3. redisplay and schedule event

But suppose I scheduled the event, and then destroyed the frame. When the event will be handled, it will try to update the frame state before checking for valid state. Do we really want that? or should we check for valid state before doing anything in the method?

In code, should I get in trouble If I do this:
```lisp
;;; After loading clim-examples

(let ((frame (clim:make-application-frame 'clim-demo::graph-toy)))
  (clim-sys:make-process 
   (lambda () (clim:run-frame-top-level frame)))
  (sleep 0.3)
  (clim:destroy-frame frame)
  (setf (clim-demo::val-array frame) nil))
```

Reply:
I felt what I did was a reasonable solution, but since this is an example program I guess the root problem should be solved, not the bug in this particular program.

I personally think that scheduled events should be cancelled when the frame is destroyed but I'm not familiar enough with the event system to know for sure. Does that sound like a reasonable solution?

Reply:
That is reasonable.